database/postgresql/src/main/java/com/example/javaopencommerce/catalog/CatalogQueryRepositoryImpl.java

```diff
@@ -4,8 +4,10 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
+import javax.enterprise.context.ApplicationScoped;
 import java.util.List;
 
+@ApplicationScoped
```

Comment:
Not a fan of that annotation. It would be better to create bean with a `@Producer`.